tools/release/CMakeLists.txt

```diff
@@ -8,4 +10,55 @@ if(NOT CMAKE_HOST_WIN32 AND PLATFORM_ARCH)
             execute_process(COMMAND ${CMAKE_CURRENT_BINARY_DIR}/prepareRelease.sh)
         ")
     endif()
+elseif(APPLE)
+    if(CMAKE_CROSSCOMPILING)
+        # copy Frameworks
+        foreach(lib IN LISTS SDL_LIBRARY SDL2_LIBRARY SDL_MIXER_LIBRARY)
+            if(lib MATCHES "\\.framework$")
+                install(DIRECTORY ${lib}
+                    DESTINATION ${RTTR_MACOS_DIR}/../Frameworks
+                    PATTERN "Headers" EXCLUDE
+                )
+            endif()
+        endforeach()
+        # copy miniupnpc
+        get_filename_component(MINIUPNPC_DIR ${MINIUPNPC_LIBRARY} DIRECTORY)
+        get_filename_component(MINIUPNPC_NAME ${MINIUPNPC_LIBRARY} NAME_WE)
+        file(GLOB MINIUPNPC_DYLIBS ${MINIUPNPC_DIR}/${MINIUPNPC_NAME}*.dylib)
+        foreach(lib IN LISTS MINIUPNPC_DYLIBS)
+            install(FILES ${lib} DESTINATION ${RTTR_LIBDIR})
+        endforeach()
+    else()
+        set(APP "\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}")
+        set(DIRS "${CMAKE_BINARY_DIR}")
+        # Path used for searching by FIND_XXX(), with appropriate suffixes added
+        if(CMAKE_PREFIX_PATH)
+            foreach(dir ${CMAKE_PREFIX_PATH})
+                list(APPEND DIRS "${dir}/bin" "${dir}/lib")
+            endforeach()
+        endif()
+        install(CODE "
+            include(BundleUtilities)
+            set(BU_CHMOD_BUNDLE_ITEMS ON)
+            file(GLOB_RECURSE LIBS \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${RTTR_DRIVERDIR}/*.*)
+            fixup_bundle(\"${APP}\" \"\${LIBS}\" \"${DIRS}\")"
+        )
+    endif()
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" OR CYGWIN)
+    install(PROGRAMS "bin/rttr.sh" DESTINATION "${RTTR_BINDIR}")
+    # copy miniupnpc
+    message(STATUS "MINIUPNPC_LIBRARY ${MINIUPNPC_LIBRARY}")
+    get_filename_component(MINIUPNPC_DIR ${MINIUPNPC_LIBRARY} DIRECTORY)
+    find_library(MINIUPNPC_BIN NAMES miniupnpc.so libminiupnpc.so HINTS ${MINIUPNPC_DIR})
+    if(MINIUPNPC_BIN)
+        get_filename_component(MINIUPNPC_BIN_RES ${MINIUPNPC_BIN} REALPATH)
+        install(FILES ${MINIUPNPC_BIN} DESTINATION ${RTTR_LIBDIR})
```

Comment:
> the so should not be necessary, but the other one (the .so is only the library name to be linked against)

True. Removed it.

> we could instead use ldd or objdump to determine all necessary dependencies. like the mingw variant in the prepareRelease.sh (which I wanted to be replaced with cmake?)

This is yet to come. Once I got the time I'll use the bundle-utilities to do that, but for now we need to get it working. Note that I organized everything here in preparation for that: This CMakeLists can later call the bundle-utils and most of this will get removed. This is why I moved the OSX stuff here too.

Reply:
okay you see it as an intermediate state, okay